timemachine/fe/rbfe.py

```diff
@@ -181,13 +174,28 @@ def generate_samples(st, host_config, temperature, lambda_schedule, n_frames, se
         group_idxs = get_group_indices(get_bond_list(hgs.bond))
         run_seed = seed + lamb_idx
         combined_masses = np.concatenate([host_masses, st.combine_masses()])
-        frames, boxes = sample(
-            bound_impls, combined_masses, temperature, x0, v0, box0, lamb, group_idxs, run_seed, n_frames=n_frames
-        )
+        dt = 1e-3
+        friction = 1.0
+        intg = LangevinIntegrator(temperature, dt, friction, combined_masses, run_seed)
+        baro = MonteCarloBarostat(len(combined_masses), 1.0, temperature, group_idxs, 15, run_seed + 1)
+        state = InitialState(U_fns, intg, baro, x0, v0, box0, lamb)
+        initial_states.append(state)
+
+    return initial_states
+
+
+def generate_samples_for_all_states(initial_states, protocol):
+    all_frames, all_boxes = [], []
+    for state_idx, state in enumerate(initial_states):
+        try:
+            frames, boxes = sample(state, protocol)
+        except Exception as exc:
+            # propagate the line information
+            raise ValueError(f"generate_samples_for_all_states failed on {state_idx}") from exc
```

Comment:
@mcwitt not sure if this is the most "pythonic" way of adding more information to an existing exception

Reply:
I'm not sure how to do this still, I want the subprocess behavior to be:

1) Return diagnostic information (InitialState) along with the original exception if the subprocess fails on the estimate (but not on the initial state generation)
2) Catch and store the initial state somewhere for this special type of exception

Reply:
(may best to offline this 2mrw)